src/protocol/xkb.rs

```diff
@@ -8336,16 +8347,17 @@ pub struct SetNamesAuxBitcase8 {
     pub n_levels_per_type: Vec<u8>,
     pub kt_level_names: Vec<xproto::Atom>,
 }
-impl Serialize for SetNamesAuxBitcase8 {
-    type Bytes = Vec<u8>;
-    fn serialize(&self) -> Self::Bytes {
+impl SetNamesAuxBitcase8 {
+    fn serialize(&self, n_types: u8) -> Vec<u8> {
         let mut result = Vec::new();
-        self.serialize_into(&mut result);
+        self.serialize_into(&mut result, n_types);
         result
     }
-    fn serialize_into(&self, bytes: &mut Vec<u8>) {
+    fn serialize_into(&self, bytes: &mut Vec<u8>, n_types: u8) {
```

Comment:
What do you think about this? Structures that have external parsing parameters will also receive those parameters in their serialize function in order to check for consistency. Those structs will not implement `Serialize` any longer.

Reply:
From the example you picked, this seems to just "shuffle" some code around, i.e. move the `assert` from the caller to the callee.
However, from other code in the diff, it seems like this actually makes some `assert`s possible that were not possible before. I am all for defensively using asserts to ensure stuff is correct, so :+1: